torch/testing/_internal/common_modules.py

```diff
@@ -158,6 +158,38 @@ def formatted_name(self):
         return self.name.replace('.', '_')
 
 
+def _zero_batch_dim_input(module_input):
+    def _zero_first_dim(objs):
+        if isinstance(objs, FunctionInput):
+            return FunctionInput(*_zero_first_dim(objs.args), **_zero_first_dim(objs.kwargs))
+        elif isinstance(objs, (list, tuple)):
+            return tuple(_zero_first_dim(obj) for obj in objs)
+        elif isinstance(objs, dict):
+            return {name: _zero_first_dim(obj) for name, obj in objs.items()}
+        elif isinstance(objs, torch.Tensor):
+            new_shape = (0, *objs.shape[1:])
+            return objs.new_empty(new_shape, requires_grad=objs.requires_grad)
+        else:
+            return objs
+
+    return ModuleInput(
+        constructor_input=_zero_first_dim(module_input.constructor_input),
+        forward_input=_zero_first_dim(module_input.forward_input),
+        desc=module_input.desc + 'zero_batch_dim')
+
+def _module_inputs_with_zero_batch_dim(f):
+    # Creates ModuleInput by reducing the size of the 0th dimension to 0
+    # in constructor_input and forward_input of module_inputs[0]
+    # Make sure the tensors in module_inputs[0] semantically make sense with
+    # the 0th dimension set to size 0
```

Comment:
So this imposes a restriction on the `module_inputs_func` that the first `ModuleInput` needs to have a batch dim so this can zero it out?

I think it'd be a little less strange if the function applied directly to the first input. Maybe something like `duplicate_with_zero_batch_dim_input(module_input) -> [Tensor]` idk - wdyt?

Reply:
I updated the PR with a version where the function is applied directly to the first input and appended to `module_inputs` in `module_inputs_func`.